pkg/cmd/ssh-key/add/http.go

```diff
@@ -3,21 +3,50 @@ package add
 import (
 	"bytes"
 	"encoding/json"
+	"fmt"
 	"io"
 	"net/http"
+	"strings"
 
 	"github.com/cli/cli/v2/api"
 	"github.com/cli/cli/v2/internal/ghinstance"
+	"github.com/cli/cli/v2/pkg/cmd/ssh-key/shared"
+	"github.com/cli/cli/v2/pkg/cmdutil"
+	"github.com/cli/cli/v2/pkg/iostreams"
 )
 
-func SSHKeyUpload(httpClient *http.Client, hostname string, keyFile io.Reader, title string) error {
+func SSHKeyUpload(httpClient *http.Client, hostname string, keyFile io.Reader, title string, printSuccessMsgs bool, ios *iostreams.IOStreams) error {
 	url := ghinstance.RESTPrefix(hostname) + "user/keys"
+	cs := ios.ColorScheme()
 
 	keyBytes, err := io.ReadAll(keyFile)
 	if err != nil {
 		return err
 	}
 
+	userKey := string(keyBytes)
+	splitKey := strings.Fields(userKey)
+	if len(splitKey) < 2 {
+		fmt.Fprintf(ios.ErrOut, "%s Error: provided key is not in a valid format\n", cs.FailureIcon())
+		return cmdutil.SilentError
+	}
+
+	userKey = splitKey[0] + " " + splitKey[1]
```

Comment:
Do you mean why reuse `userKey` instead of making a new variable? I could do that, then I could pass `userKey` below instead of having to call `string(keyBytes)` again, that makes more sense. I'll update that.

Reply:
I meant that userKey should already have this value (ie pre-split), so line 34 here is a no-op (it seems)

Reply:
Oh, I have to do this because the API returns keys in the format `ssh-ed25519 keyHere`, but the user's key file may have extra fields after that (could be something like `ssh-ed25519 keyHere keyNameHere`. I have to strip out just the part that the API returns to compare the strings.

Reply:
OH my brain inserted `strings.Split` but I see you're calling `strings.Fields` 🤦‍♀️ I totally understand now, sorry